ui/plugin-system/src/runtime/datasources.ts

```diff
@@ -53,3 +53,11 @@ export function useListDatasources(datasourcePluginKind: string) {
   const { listDatasourceMetadata } = useDatasourceStore();
   return useQuery(['listDatasourceMetadata', datasourcePluginKind], () => listDatasourceMetadata(datasourcePluginKind));
 }
+
+/**
+ * Provides a convience hook for getting a DatasourceClient for a given DatasourceSelector.
+ */
+export function useDatasourceClient<Client>(selector: DatasourceSelector) {
```

Comment:
I added this new plugin API to make it easy to load the client datasource from a React component

Reply:
Perfect 👍